src/main/java/org/apache/pulsar/io/jcloud/sink/BlobStoreAbstractSink.java

```diff
@@ -266,15 +286,27 @@ private void unsafeFlush() {
                 return;
             }
 
-            String filepath = "";
+            String filepath;
+            try {
+                if (partitioner instanceof LegacyPartitioner) {
+                    filepath = buildPartitionPath(firstRecord, legacyPartitioner, format, timeStampForPartitioning);
```

Comment:
https://github.com/streamnative/pulsar-io-cloud-storage/pull/845/files#diff-ceb9cd48b8d0b77350207c635b5695442fc0ade9f6c486fc6f31efd432a2d078R357-R366

Can we abstract this method to legacyPartitioner? This puts the legacy code together to make this sink code clearer.